CMakeLists.txt

```diff
@@ -1,6 +1,20 @@
 cmake_minimum_required(VERSION 3.16)
 project(SIM2022)
 
+if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
+  if(CMAKE_CXX_COMPILER_VERSION VERSION_LESS "10.3.0")
+    message(FATAL_ERROR "Insufficient gcc version")
+  endif()
+elseif(CMAKE_CXX_COMPILER_ID STREQUAL "MSVC")
+  if(CMAKE_CXX_COMPILER_VERSION VERSION_LESS "14.32.31326")
+    message(FATAL_ERROR "Insufficient msvc version")
+  endif()
+elseif(CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
+  if(CMAKE_CXX_COMPILER_VERSION VERSION_LESS "14.0.0")
+    message(FATAL_ERROR "Insufficient msvc version")
+  endif()
+endif()
```

Comment:
Could you please add more extended error message with info about current compiler version?
In this case it may be useful to write a special macro

Reply:
I think that you misspelled

Reply:
> Could you please add more extended error message with info about current compiler name & version? In this case it may be useful to write a special macro

I think that you misspelled
 `  if(CMAKE_CXX_COMPILER_VERSION VERSION_LESS "14.0.0")
    message(FATAL_ERROR "Insufficient msvc version")`
should be 
`message(FATAL_ERROR "Insufficient clang version")`

Reply:
It is one of the reason implement macro)
> > Could you please add more extended error message with info about current compiler name & version? In this case it may be useful to write a special macro
> 
> 
> 
> I think that you misspelled
> 
>  `  if(CMAKE_CXX_COMPILER_VERSION VERSION_LESS "14.0.0")
> 
>     message(FATAL_ERROR "Insufficient msvc version")`
> 
> should be 
> 
> `message(FATAL_ERROR "Insufficient clang version")`

Reply:
Done